src/deploy.ts

```diff
@@ -318,7 +318,8 @@ export class MBDeployer implements MBDeployerI {
    */
   private async linkContractToAddress(
     contract: MultiBaasContract,
-    address: MultiBaasAddress
+    address: MultiBaasAddress,
+    startingBlock: string
```

Comment:
I got 500 status when the block number is 2 (or more) blocks ahead the current block.

_MultiBaas (local)_
```sh
INFO[2022-06-20 12:32:03.070648+0900] API call result: /api/v0/chains/ethereum/addresses/sample_nft/contracts  api_key=2 forwarded-for= internal="not found" message="an internal error occurred" method=POST remoteaddr="127.0.0.1:51718" status=500 user=demo@curvegrid.com
```

_Plugin_
```sh
Error: Request failed with status code 500
    at createError (/Users/daenamkim/Developments/workspace/curvegrid/PUBLIC/hardhat-multibaas-plugin/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/Users/daenamkim/Developments/workspace/curvegrid/PUBLIC/hardhat-multibaas-plugin/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/Users/daenamkim/Developments/workspace/curvegrid/PUBLIC/hardhat-multibaas-plugin/node_modules/axios/lib/adapters/http.js:322:11)
    at IncomingMessage.emit (node:events:402:35)
    at endReadableNT (node:internal/streams/readable:1343:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  config: {
    transitional: {
      silentJSONParsing: true,
...

```

I found this: 

https://github.com/curvegrid/multibaas/blob/b184a5b43d8d39a9b59b433bbb4491c08309346c/server/app/web/api/v0/addresses-api_test.go#L1397-L1403

it seems intended but what about to add `try/catch` to show error message like `Block does not exist` or `Block should not be bigger than the current block (xx)`

Reply:
My bad, I didn't see this during review. We should change to a 400 error. I will file an issue.

Reply:
I have added https://github.com/curvegrid/multibaas/issues/3994 on behalf of you.